### ISSUE TITLE
window.print() prints only once

### ISSUE THREAD

**Issue by daumling** [electron/electron#21195]
### Preflight Checklist

* [ X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [ X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [ X] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:** 
  * 7.1.1 
* **Operating System:** 
  * macOS 10.15.1, Windows 10 (1903) 
* **Last Known Working Electron version:**
  * 

### Expected Behavior

window.print() should print every time it is called
### Actual Behavior

After the first printing operation, window.print() displays the print dialog, but does not produce any output; no printing, no saving as PDF, no opening in Preview, just nothing.
### To Reproduce

[Gist](https://gist.github.com/f10bd81922f43f0495)

Click "Print Me" twice.

On the first click, the window is printed. On further clicks, the print dialog appears, but no output is generated.

**Comment by t57ser** [comment]
Can confirm. Seeing the same in 7.0.0 / 7.1.0

**Comment by baconbrad** [comment]
The gist is a broken link. I cannot replicate this in my project using 7.1.1. You mentioned a dialog but I am using silent printing so that may be a factor if it is broke.

**Comment by TerrapinSoftware** [comment]
[New Gist](https://gist.github.com/8feedbc56b502b62e9393decd18ed771)

In case the link is broken as well, just add this HTML to index.html:

```html
<button onclick="window.print()">Print me</button>
```
And click twice. First time, it prints, second time, it does not after displaying the print dialog correctly.

**Comment by InfiniteLukeOne** [comment]
Having the same problem in my project using 7.1.1. First time printing works, second time and each next time only print options dialog is displayed but when confirming nothing gets printed.

**Comment by bjbk** [comment]
I switched to using ipc for printing.  I don't see the same issue with this in main:
```js
// main.js
ipcMain.on('print', (e) => {
  mainWindow.webContents.print()
})

// renderer
print () {
   ipcRenderer.send('print')
 }
```

**Comment by TerrapinSoftware** [comment]
Unfortunately, this does not work for iframes. It is possible to find the iframe webContents ID and send it to main via `parent.remote.getCurrentWebContents().id` (if you set `remote`in the top level code of the renderer process), but this code in main.js:

```
ipcMain.on('print', (evt, id) => {
   let wc = webContents.fromId(id);
  if (wc)
    wc.print();
});
```
causes Electron to hang seriously after displaying the print dialog.

**Comment by vladimiry** [comment]
@TechnicalSoup 
>  It is possible to find the iframe webContents ID

> ipcMain.on('print', (evt, id) => {

Try using `evt.sender` prop as its supposed to be the `WebContents` instance that sent an event to the main process. So the printing  call would belooking like `evt.sender.print()`.

**Comment by StefanPlehn** [comment]
@TerrapinSoftware @vladimiry 
I tried both of your solutions and none of them worked for an iframe.
Maybe it's because I'm not sure what you mean by
> It is possible to find the iframe webContents ID and send it to main via `parent.remote.getCurrentWebContents().id` (if you set remotein the top level code of the renderer process)
Do you have some example code what to do in the JavaScript Code and what in the main.js?
That would be great.
Thank You.

**Comment by vladimiry** [comment]
@StefanPlehn , using `parent.remote.getCurrentWebContents().id` approach is in general a bad practice. The good practice is to use a pure events-based IPC/inter-process communication approach only. There is no need to use `parent.remote.getCurrentWebContents().id`-like code if you follow the way I highlighted above (I didn't run this code but it's supposed to work):
- Setup a handler in the `main` process:
```
    ipcMain.on("print", (event) => {
        event.sender.print();
    });
```
- Then trigger that event from `renderer` process:
```
ipcRenderer.send("print")
```

**Comment by TerrapinSoftware** [comment]
I'd love to try - even if it means that I need to distribute a new executable. But how am I supposed to get access to ipcRenderer inside an iframe? Any suggestions would be greatly appreciated - after all, there is no require('electron') in web content loaded in an iframe...

**Comment by vladimiry** [comment]
@TerrapinSoftware the `print` method is available on WebContents instance in the main process. So you could, for example, send HTML data that needs to be printed to dynamically created and hidden BrowserWindow/BrowserView/WebView instance. You receive the HTML data there, render it and call `ipcRenderer.send("print")`. Doing rendering in that hidden BrowserWindow/BrowserView/WebView instance you could use iframe if needed. I guess such logic could be wrapped in a library.

**Comment by TerrapinSoftware** [comment]
This workaround works great, thank you! Even for the main window. I still think that the bug should be fixed...

**Comment by codebytere** [comment]
Closing as a duplicate of https://github.com/electron/electron/issues/21397, which i've just opened a PR to fix!

**Comment by Thalinda** [comment]
This one is FIXED in the Electron 8.00

**Comment by dpeckham** [comment]
I have this issue in 11.1.1.

**Comment by davidjgraph** [comment]
11.1.1 is pretty old and pre-dates Electron 8. You need to update.
